scripts/ci/print-instrumentation-failures.py

```
@@ -0,0 +1,119 @@
+
+
+def find_result_xml(results_dir):
+    """結果ディレクトリ配下の JUnit XML を再帰的に集める。"""
+    return sorted(glob.glob(os.path.join(results_dir, "**", "*.xml"), recursive=True))
+
+
+def collect_failures(xml_path):
+    """1 つの JUnit XML から failure / error のテストケースを抽出する。"""
+    failures = []
+    try:
+        root = ET.parse(xml_path).getroot()
```

Ruff (0.15.14)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)